crates/router/src/connector/boku/transformers.rs

```diff
@@ -112,47 +268,96 @@ impl<F, T>
+        None => Err(errors::ConnectorError::MissingConnectorRedirectionPayload {
+            field_name: "redirect_url",
+        }),
+    }?;
+
+    Ok((status, response.charge_id, redirection_data))
+}
+
+fn get_psync_response(
+    response: BokuPsyncResponse,
+) -> Result<(enums::AttemptStatus, String, Option<RedirectForm>), errors::ConnectorError> {
+    let status = get_response_status(response.charges.charge.charge_status);
+
+    Ok((status, response.charges.charge.charge_id, None))
+}
+
 // REFUND :
-#[derive(Default, Debug, Serialize)]
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename = "refund-charge-request")]
 pub struct BokuRefundRequest {
-    pub amount: i64,
+    refund_amount: i64,
+    merchant_id: Secret<String>,
+    merchant_request_id: String,
+    merchant_refund_id: Secret<String>,
+    charge_id: String,
+    reason_code: String,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub enum BokuRefundReasonCode {
+    NonFulfillment,
+}
+
+impl fmt::Display for BokuRefundReasonCode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::NonFulfillment => write!(f, "8"),
+        }
+    }
 }
 
 impl<F> TryFrom<&types::RefundsRouterData<F>> for BokuRefundRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
-        Ok(Self {
-            amount: item.request.refund_amount,
-        })
+        let auth_type = BokuAuthType::try_from(&item.connector_auth_type)?;
+        let payment_data = Self {
+            refund_amount: item.request.payment_amount,
+            merchant_id: auth_type.merchant_id,
+            merchant_refund_id: Secret::new(item.payment_id.to_string()),
+            merchant_request_id: Uuid::new_v4().to_string(),
+            charge_id: item.request.connector_transaction_id.to_string(),
+            reason_code: BokuRefundReasonCode::NonFulfillment.to_string(),
+        };
+
+        Ok(payment_data)
     }
 }
 
-// Type definition for Refund Response
-#[allow(dead_code)]
-#[derive(Debug, Serialize, Default, Deserialize, Clone)]
-pub enum RefundStatus {
-    Succeeded,
-    Failed,
-    #[default]
-    Processing,
+//TODO: Fill the struct with respective fields
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename = "refund-charge-response")]
+pub struct RefundResponse {
+    charge_id: String,
+    refund_status: String,
 }
 
-impl From<RefundStatus> for enums::RefundStatus {
-    fn from(item: RefundStatus) -> Self {
-        match item {
-            RefundStatus::Succeeded => Self::Success,
-            RefundStatus::Failed => Self::Failure,
-            RefundStatus::Processing => Self::Pending,
-        }
+fn get_refund_status(status: String) -> enums::RefundStatus {
+    match status.as_str() {
+        "Success" => enums::RefundStatus::Success,
+        "Failure" => enums::RefundStatus::Failure,
+        _ => enums::RefundStatus::TransactionFailure,
```

Comment:
Same here, Can we have separate enum for all the status?

Reply:
quick xml parse differently than serde json, to have correct formatting cant provide enum